test/jdk/javax/swing/JTabbedPane/bug4287268.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2000, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4287268
+ * @summary Tests if setIconAt(index,Icon) does not set Tab's disabled icon
+ * @key headful
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTabbedPane;
+import javax.swing.SwingUtilities;
+
+public class bug4287268 {
+
+    static JFrame frame;
+    static volatile JTabbedPane jtp;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(bug4287268::createUI);
+
+            Robot robot = new Robot();
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            Point point = jtp.getLocationOnScreen();
+            int width = jtp.getWidth();
+            int height = jtp.getHeight();
```

Comment:
Might be good to wrap it in EDT. 

It was the following `getComponentBounds()` method that @azvegint suggested we move to Util (since we use across multiple tests), but it does create a dependency on Util class.

```
Rectangle jtpLoc = Util.invokeOnEDT(() ->
                                        getComponentBounds(jtp));

    private static Rectangle getComponentBounds(Component component) {
            Point locationOnScreen = component.getLocationOnScreen();
            Dimension size = component.getSize();
            return new Rectangle(locationOnScreen, size);
    }
```

Reply:
As I said when it was  discussed, I like to avoid dependencies just for the sake of a few lines of code.

Reply:
Yes, I agree it does create the extra dependency if Util is used.
Without using Util, is it good to have JTabbedPane's location and size on EDT ?